### PR TITLE
vRA 8.x API upgrade

### PR DESCRIPTION
**Pull request by ashiqueps**
Signed-off-by: Ashique Saidalavi <ashique.saidalavi@progress.com>

### Description

Added support for the vRA 8.x

### Issues Resolved

#83 

### Check List

- [x] All tests pass.
- [x] All style checks pass.
- [x] Functionality includes testing.
- [x] Functionality has been documented in the README if applicable

### CODE REVIEW

File: lib/vra/catalog_base.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+#
+# Author:: Ashique Saidalavi (<ashique.saidalavi@progress.com>)
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
```
Comment (vkarve-chef) [inline lib/vra/catalog_base.rb:4]: copyright year could be updated to 2021

File: spec/deployment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,227 @@
+# frozen_string_literal: true
+#
+# Author:: Chef Partner Engineering (<partnereng@chef.io>)
```
Comment (vkarve-chef) [inline spec/deployment_spec.rb:3]: author could be updated in the new specs.

**Review by vkarve-chef — APPROVED** [review]

File: lib/vra/catalog_base.rb (hunk 1 above)
Comment (ashiqueps) [reply lib/vra/catalog_base.rb]: Done

File: lib/vra/client.rb [hunk 3]
```diff
@@ -43,71 +47,76 @@ def initialize(opts)
...
 
-      response = http_post("/identity/api/tokens",
-                           FFI_Yajl::Encoder.encode(bearer_token_request_body),
+      response = http_post(ACCESS_TOKEN_URL,
+                           FFI_Yajl::Encoder.encode(token_params),
                            :skip_auth)
-      unless response.success_ok?
-        raise Vra::Exception::Unauthorized, "Unable to get bearer token: #{response.body}"
-      end
+      raise Vra::Exception::Unauthorized, "Unable to get bearer token: #{response.body}" unless response.success_ok?
```
[the remark this answers is not in the document]
Comment (ashiqueps) [reply lib/vra/client.rb]: Missed it. Updated the message.

File: spec/deployment_spec.rb (hunk 2 above)
Comment (ashiqueps) [reply spec/deployment_spec.rb]: Done

**Review by damacus — APPROVED** [review]

**Comment by damacus** [comment]
👏🏼 that's some really nice code there. Thanks for making it so easy to review.

**Pull request merged**